db/migrations/20190723233131_test_change_type.cr

```diff
@@ -0,0 +1,35 @@
+class TestChangeType::V20190723233131 < Avram::Migrator::Migration::V1
+  # Test that we can save and query after type changes
+  class TempUserInt64 < Avram::Model
+    skip_default_columns
+
+    def self.database
+      TestDatabase
+    end
+
+    table :users do
+      primary_key id : Int64
+      column name : String
+      column age : Int32
+      column joined_at : Time
+      timestamps
+    end
+  end
+
+  def migrate
+    TempUserInt64::SaveOperation.create!(name: "foo", age: 0, joined_at: Time.now)
+
+    alter table_for(User) do
+      change_type id : Int32
+    end
+
+    alter table_for(User) do
+      change_type id : Int64
+    end
+
+    TempUserInt64::BaseQuery.first # should not raise
```

Comment:
I think this should cover it. We change it and then back and then run a query that would raise if the row is missing or id was not changed correctly